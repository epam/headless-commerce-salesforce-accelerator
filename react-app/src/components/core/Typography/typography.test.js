/*
 * Copyright 2020 EPAM Systems, Inc. (https://www.epam.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import { shallow } from "enzyme";

import StyledTypography from "./typographyStyled";
import getMainTheme from "../../../theme";

const defaultProps = {
  fontSize: 14,
  lineheight: 24,
  margin: "8px 0 0 0",
  colortheme: "mineShaft",
  fontWeight: "regular",
  align: "right",
  display: "block",
};

const theme = getMainTheme();

describe("StyledTypography", () => {
  it("it should match snapshot", () => {
    const wrapper = shallow(
      <StyledTypography theme={theme} {...defaultProps} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
