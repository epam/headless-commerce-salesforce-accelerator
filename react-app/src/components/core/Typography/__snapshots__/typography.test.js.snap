// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledTypography it should match snapshot 1`] = `
<WithStyles(ForwardRef(Typography))
  align="right"
  className="sc-AxjAm fOICLa"
  colortheme="mineShaft"
  display="block"
  fontSize={14}
  fontWeight="regular"
  lineheight={24}
  margin="8px 0 0 0"
  theme={
    Object {
      "breadcrumbsH": 40,
      "breakpoints": Object {
        "mobile": 992,
      },
      "colLg": 74,
      "colMd": 69,
      "colSm": 60,
      "colors": Object {
        "alabaster": "#FBFBFB",
        "amber": "#FFC000",
        "athensGray": "#FAFBFC",
        "black": "#000",
        "blue": "#376A9A",
        "dimGray": "#CBCDCE",
        "gullGrey": "#A7A9AA",
        "iron": "#D9DCDD",
        "lightGray": "#DEE2E3",
        "mineShaft": "#222222",
        "miscellaneous": "linear-gradient(90deg, #E77878 0%, #EFD063 25.00%, #65D5EE 50.00%, #C083E6 75.00%, #7F6EE8 100%);",
        "pale": "#FFEFEF",
        "punch": "#DA3232",
        "regentGrey": "#7D8081",
        "rollingStone": "#7E8182",
        "semiWhite": "rgba(255, 255, 255, 0.75)",
        "silverChalice": "#A9A9A9",
        "turmeric": "#C5D153",
        "wattle": "#CEDB56",
        "whisper": "#EBEBEB",
        "white": "#FFFFFF",
        "yellowGreen": "#D8E278",
      },
      "contentWidth": 1200,
      "font": [Function],
      "fontSize": 14,
      "footerH": 331,
      "gutter": 24,
      "headerH": 160,
      "navBarH": 56,
      "pxToRem": [Function],
      "searchBarH": 72,
      "topBarH": 32,
    }
  }
/>
`;
